compile.sh

```diff
@@ -4,62 +4,105 @@
 # to cross-compile for different architectures on your host machine.
 # http://dave.cheney.net/2015/08/22/cross-compilation-with-go-1-5
 
+# ensure the compiler exits with status 0
+function assert() {
```

Comment:
We could use `-e` which will cause the bash script to exit with failure if any line fails. see [bash unofficial strict mode](http://redsymbol.net/articles/unofficial-bash-strict-mode/). Just change the first line of this file to `#!/bin/bash -e` and you can remove all usage of assert. If any individual line should be allowed to fail you can turn it off temporarily.